201-storage-advanced-threat-protection-create/azuredeploy.json

```diff
@@ -0,0 +1,87 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "storageAccountName": {
+      "type": "string",
+      "defaultValue": "storagewatp1",
+      "metadata": {
+        "description": "The name must be unique across all existing storage account names in Azure. It must be 3 to 24 characters long, and can contain only lowercase letters and numbers."
+      }
+    },
+    "storageAccountLocation": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Storage account location, default is same as resource group location."
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "StorageV2",
+      "allowedValues": [
+        "StorageV2",
+        "Storage"
+      ],
+      "metadata": {
+        "description": "Storage Account type, for more info see 'https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview'."
+      }
+    },
+    "storageAccountReplication": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS",
+        "Premium_LRS"
+      ],
+      "metadata": {
+        "description": "Storage Account replication, for more info see 'https://docs.microsoft.com/en-us/azure/storage/common/storage-redundancy'."
+      }
+    },
+    "advancedThreatProtectionEnabled": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Enable or disable Advanced Threat Protection."
+      }
+    }
+  },
+  "variables": {
+    "storageAccountName": "[parameters('storageAccountName')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "location": "[parameters('storageAccountLocation')]",
+      "apiVersion": "2018-07-01",
+      "sku": {
+        "name": "[parameters('storageAccountReplication')]"
+      },
+      "kind": "[parameters('storageAccountType')]",
+      "properties": {},
+      "resources": [
+        {
+          "condition": "[parameters('advancedThreatProtectionEnabled')]",
+          "type": "providers/advancedThreatProtectionSettings",
+          "name": "Microsoft.Security/current",
+          "apiVersion": "2017-08-01-preview",
+          "dependsOn": [
+            "[concat('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]"
+          ],
+          "properties": {
+            "isEnabled": true
```

Comment:
parameters('advancedThreatProtectionEnabled')?

Reply:
this call would happen only when `"condition": "[parameters('advancedThreatProtectionEnabled')]"` is `true`,
as it shouldn't be called at all rather then be called with `false`.